packages/node/src/indexer/apiPromise.connection.ts

```diff
@@ -39,43 +47,62 @@ export class ApiPromiseConnection implements ApiConnection {
       ...args.chainTypes,
     };
     const api = await ApiPromise.create(apiOption);
-    return new ApiPromiseConnection(api);
+    return new ApiPromiseConnection(api, fetchBlocksBatches);
   }
 
-  get api(): ApiPromise {
-    return this._api;
+  async fetchBlocks(
+    heights: number[],
+    overallSpecVer?: number,
+  ): Promise<BlockContent[]> {
+    const blocks = await this.fetchBlocksBatches(
+      this.unsafeApi,
+      heights,
+      overallSpecVer,
+    );
+    return blocks;
   }
 
   async apiConnect(): Promise<void> {
-    await this._api.connect();
+    await this.unsafeApi.connect();
```

Comment:
`IApi` seems to have connect and disconnect methods. I don't think it makes sense to use `unsafeApi` here. It implies that it is a different instance to `safeApi`